packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -0,0 +1,141 @@
+// found in the LICENSE file.
+
+import 'package:vector_math/vector_math_64.dart';
+
+import 'object.dart';
+import 'sliver.dart';
+
+/// A sliver that places multiple sliver children in a linear array along the cross
+/// axis.
+///
+/// Since the extent of the viewport in the cross axis direction is finite,
+/// each sliver is forced to divide up the extent per specification via
+/// [ParentDataWidget] [SliverConstrainedCrossAxis] and [SliverExpanded].
+///
+/// The algorithm for dividing up the cross axis extent is as follows.
+/// Every widget either has constrained cross axis extent or a flex value.
+/// First allocate space to all the widgets with constrained cross axis extent.
+/// Then divide up the remaining space proportional to the widget's flex value.
+/// Flex values can be specified via the [SliverExpanded] widgets, but if
+/// a sliver without [SliverExpanded] or [SliverConstrainedCrossAxis] is passed,
+/// it is assumed to have a flex value of 1, which by itself expands
+/// to take up all of the cross axis extent.
+class RenderSliverCrossAxisGroup extends RenderSliver with ContainerRenderObjectMixin<RenderSliver, SliverPhysicalContainerParentData> {
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! SliverPhysicalContainerParentData) {
+      child.parentData = SliverPhysicalContainerParentData();
+    }
+  }
+
+  @override
+  double childMainAxisPosition(RenderSliver child) => 0.0;
+
+  @override
+  double childCrossAxisPosition(RenderSliver child) {
+    switch (constraints.axisDirection) {
+      case AxisDirection.up:
+      case AxisDirection.down:
+        return (child.parentData! as SliverPhysicalContainerParentData).paintOffset.dx;
+      case AxisDirection.left:
+      case AxisDirection.right:
+        return (child.parentData! as SliverPhysicalContainerParentData).paintOffset.dy;
+    }
+  }
+
+  @override
+  void performLayout() {
+    // Iterate through each sliver.
+    // Get the parent's dimensions.
+    final double crossAxisExtent = constraints.crossAxisExtent;
+    assert(crossAxisExtent.isFinite);
+
+    // First, layout each child with flex == 0.
+    int totalFlex = 0;
+    double usedUpExtent = 0.0;
+    RenderSliver? child = firstChild;
+    while (child != null) {
+      final SliverPhysicalContainerParentData childParentData = child.parentData! as SliverPhysicalContainerParentData;
+      final int flex = childParentData.flex ?? 1;
+      if (flex == 0) {
+        child.layout(constraints, parentUsesSize: true);
+        usedUpExtent += child.geometry!.crossAxisExtent;
+        print(usedUpExtent);
+      } else {
+        totalFlex += flex;
+      }
+      child = childAfter(child);
+    }
+    final double extentPerFlexValue = (crossAxisExtent - usedUpExtent) / totalFlex;
+
+    child = firstChild;
+    double offset = 0.0;
+
+    // At this point, all slivers with constrained cross axis should already be laid out.
+    geometry = SliverGeometry.zero;
```

Comment:
Why is the geometry of the group set to 0 here?

Reply:
To find the maximum scrollExtent sliver, we initially set it to zero, and update it with geometries with higher scrollExtents.